tpot/base.py

```diff
@@ -238,6 +252,8 @@ def __init__(self, generations=100, population_size=100, offspring_size=None,
         self.early_stop = early_stop
         self._last_optimized_pareto_front = None
         self._last_optimized_pareto_front_n_gens = 0
+        self.memory = memory
+        self._memory = None # initial Memory setting for sklearn pipeline
```

Comment:
Why is there both `memory` and `_memory` attributes? Shouldn't there only be one?

Reply:
`self.memory` is parameter but `_memory` can only be Memory object. `self.memory` is needed for checking the initial parameter in the end of optimization process (if it is "auto", then clean up memory caching and temporary directory).

Reply:
Hmm, good point. I suppose `fit` shouldn't override that parameter value.